scio-jdbc/src/main/scala/com/spotify/scio/jdbc/sharded/ShardString.scala

```diff
@@ -96,4 +100,85 @@ object RangeShardStringCodec {
       def lift(str: String): Base64String = Base64String(str)
     }
 
+  implicit def sqlServerUuidShardRangeStringCodec[S <: SqlServerUuidString](
+    uuidStringBuilder: String => S
+  ): RangeShardStringCodec[S] =
+    new RangeShardStringCodec[S] {
+
+      import java.util.UUID
+
+      def encode(bigInt: BigInt): S = {
+        val bytes = Array.tabulate(16) { i =>
+          ((bigInt >> (i * 8)) & 0xff).toByte
+        }
+
+        val uuidBytes = Array(
+          bytes(0),
+          bytes(1),
+          bytes(2),
+          bytes(3),
+          bytes(4),
+          bytes(5),
+          bytes(6),
+          bytes(7),
+          bytes(9),
+          bytes(8),
+          bytes(15),
+          bytes(14),
+          bytes(13),
+          bytes(12),
+          bytes(11),
+          bytes(10)
+        )
+
+        val uuid = new UUID(
+          java.nio.ByteBuffer.wrap(uuidBytes, 0, 8).getLong,
+          java.nio.ByteBuffer.wrap(uuidBytes, 8, 8).getLong
+        )
+
+        lift(uuid.toString)
+      }
+
+      def decode(str: S): BigInt = {
+        val formattedStr = str.value.replace("-", "")
+        val bytes =
+          formattedStr.grouped(2).map(Integer.parseInt(_, 16).toByte).toArray
+
+        val reversedBytes = Array(
+          bytes(0),
+          bytes(1),
+          bytes(2),
+          bytes(3),
+          bytes(4),
+          bytes(5),
+          bytes(6),
+          bytes(7),
+          bytes(9),
+          bytes(8),
```

Comment:
According to the post, there are should be many 'groups' here. shouldn't the order reverse be
`3, 2, 1, 0, 5, 4, 7, 6, 9, 8, 15, 14, 13, 12, 11, 10` ?

Reply:
I take `0, 1, 2, 3, 4, 5, 6, 7, 9, 8, 15, 14, 13, 12, 11, 10` because they say the groups 10-15 and 8-9 are stored as big-endian and the others as little-endian. After I reverse the array and I build a Bigint giving more weight to the bytes on the left.